src/cirrus/lambda_handlers/api.py

```diff
@@ -118,20 +117,30 @@ def _results_transform(
     ]
 
 
-def daily(results: Dict[str, Any]) -> List[Dict[str, Any]]:
+def daily(results: dict[str, Any] | None) -> list[dict[str, Any]]:
     return _results_transform(results, lambda x: x.split(" ")[0], "day")
 
 
-def hourly(r1: Dict[str, Any], *rs: Dict[str, Any]) -> List[Dict[str, Any]]:
+def hourly(
+    r1: dict[str, Any] | None,
+    *rs: dict[str, Any] | None,
+) -> list[dict[str, Any]]:
     combined_results = deepcopy(r1)
     for result in rs:
-        combined_results["Rows"].extend(result.get("Rows", []))
+        if not result:
+            continue
+        if not combined_results:
+            combined_results = result
```

Comment:
should this break? If not, I think it would be better if there was a check above to see if combined_results is set and rs is non-empty and then pop the last one.

Reply:
I don't know, I did what I thought would preserve the existing behavior as best as I could reason about while also supporting the case where we have `None`s passed in to satisfy type checking. Maybe the whole thing deserves a bit more consideration. It might be better to throw an exception rather than returning `None` so we can avoid this case altogether.